pkg/deviceplugin/server.go

```diff
@@ -149,6 +160,7 @@ func (srv *server) Stop() error {
 	if srv.grpcServer == nil {
 		return errors.New("Can't stop non-existing gRPC server. Calling Stop() before Serve()?")
 	}
+	srv.state = terminating
 	srv.grpcServer.Stop()
```

Comment:
The socket file needs to be removed here. Otherwise `watchFile()` won't exit. For this `pluginSocket` has to be a part of `server`'s state.

Reply:
Interestingly in my testing it does shut down. I wonder if shutting down `grpcserver` closes the file implicitly. It isn't obvious to me from reading the [docs](https://godoc.org/google.golang.org/grpc#Server.Stop).

Could calling `os.RemoveFile()` in `Stop()` introduce any further threading considerations?

Reply:
Yep, works as you expect. Done.